app/models/fixity_check.rb

```diff
@@ -39,6 +39,21 @@ def self.find_fields
   scope :with_bag_id, ->(id) { where(bag_id: id) unless id.blank? }
   scope :with_node_id, ->(id) { where(node_id: id) unless id.blank? }
   scope :latest_only, ->(flag) { group(:node_id, :bag_id).having("created_at = max(created_at)") unless flag.blank? }
+  scope :latest_only, ->(flag) do
+    unless flag.blank?
+      joins("INNER JOIN (
+        SELECT node_id, bag_id, MAX(created_at) AS max_created_at
+        FROM #{FixityCheck.table_name}
+        GROUP BY node_id, bag_id
+        ) AS x
+        ON
+        #{FixityCheck.table_name}.bag_id = x.bag_id
+        AND
+        #{FixityCheck.table_name}.node_id = x.node_id
+        AND
+        #{FixityCheck.table_name}.created_at = x.max_created_at")
+    end
+  end
```

Comment:
I'd rather not fetch all of the records.  Your `records =` query is not going to return the latest fixity check for each bag-node pair.

The fixture data is more varied than your query illustrates.
```
[2] pry(#<RSpec::ExampleGroups::FixityCheck::ScopeLatestOnly>)> FixityCheck.order(:bag_id, :node_id, :created_at).pluck(:bag_id, :node_id, :created_at)
=> [[1, 9, Mon, 24 Oct 2016 16:55:10 UTC +00:00],
 [1, 9, Wed, 26 Oct 2016 15:55:10 UTC +00:00],
 [1, 10, Mon, 24 Oct 2016 16:55:10 UTC +00:00],
 [1, 10, Wed, 26 Oct 2016 15:55:10 UTC +00:00],
 [2, 9, Mon, 24 Oct 2016 16:55:10 UTC +00:00],
 [2, 9, Wed, 26 Oct 2016 15:55:10 UTC +00:00]]
```